presentation/src/main/java/me/androidbox/presentation/agenda/viewmodel/AgendaViewModel.kt

```diff
@@ -125,6 +127,53 @@ class AgendaViewModel @Inject constructor(
                     )
                 }
             }
+
+            AgendaScreenEvent.OnLogoutClicked -> {
+                logoutCurrentUser()
+            }
+
+            is AgendaScreenEvent.OnOpenLogoutDropDownMenu -> {
+                _agendaScreenState.update { agendaScreenState ->
+                    agendaScreenState.copy(
+                        shouldOpenLogoutDropDownMenu = agendaScreenEvent.shouldOpen
+                    )
+                }
+            }
+        }
+    }
+
+    private fun logoutCurrentUser() {
+        viewModelScope.launch {
+            when(logoutUseCase.execute()) {
+                ResponseState.Loading -> Unit /* TODO Show loading */
+
+                is ResponseState.Success -> {
+                    preferenceRepository.deleteCurrentUser()
+
+                    val eventJob = viewModelScope.launch {
+                        agendaLocalRepository.deleteAllEvents()
+                    }
+                    val taskJob = viewModelScope.launch {
+                        agendaLocalRepository.deleteAllTasks()
+                    }
+                    val reminderJob = viewModelScope.launch {
+                        agendaLocalRepository.deleteAllReminders()
+                    }
+
+                    listOf(eventJob, taskJob, reminderJob).forEach { job ->
+                        job.join()
+                    }
+
+                    _agendaScreenState.update { agendaScreenState ->
+                        agendaScreenState.copy(
+                            deletedCacheCompleted = true
+                        )
+                    }
```

Comment:
@philipplackner @MKiperszmid This is the state that is set to true after the cache as been deleted. That will trigger the `logout()` in the LaunchedEffect.  Does this look ok?

Reply:
Yes